configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ServiceOperation.java

```diff
@@ -44,39 +74,23 @@
 
     private final Map<String, Object> currentProperties;
 
-    private AdminOperation(String pid, Map<String, Object> configs, boolean keepIgnored,
+    private ServiceOperation(String pid, Map<String, Object> configs, boolean keepIgnored,
             ConfigurationAdminMBean cfgAdmMbean) {
         this.pid = pid;
         this.configs = new HashMap<>(configs);
         this.keepIgnored = keepIgnored;
         this.cfgAdmMbean = cfgAdmMbean;
 
         try {
-            currentProperties = cfgAdmMbean.getProperties(pid);
-        } catch (IOException e) {
+            currentProperties = readState(this.pid, this.cfgAdmMbean);
```

Comment:
@peterhuffer Could you please retest and confirm that this resolves the problem with lost defaults with the WCPM?